pypeit/coadd2d.py

```diff
@@ -1077,29 +1316,82 @@ def __init__(self, spec2d_files, spectrograph, par, det=1, offsets=None, weights
         kwargs_wave['wave_method'] = 'log10' if 'wave_method' not in kwargs_wave else kwargs_wave['wave_method']
         self.wave_grid, self.wave_grid_mid, self.dsamp = self.get_wave_grid(**kwargs_wave)
 
-        self.objid_bri = None
-        self.slitid_bri  = None
-        self.snr_bar_bri = None
-        if offsets is None:
-            self.objid_bri, self.slitid_bri, self.snr_bar_bri = self.get_brightest_obj(self.stack_dict['specobjs_list'], self.nslits)
-        else:
-            # TODO -- Check the input offsets list matches the science images
-            pass
+        # get offsets and weights
+        self.compute_offsets_and_weights(weights, offsets)
 
-        self.use_weights = self.parse_weights(weights)
+    def compute_offsets_and_weights(self, weights, offsets):
```

Comment:
Actually, this code was added here and is completely redundant. I also think it does not make any sense for echelon since it was not a method in echelle before. I think it would be more efficient to chat.

Reply:
Just add a todo that we will break this up into two methods compute_offsets and compute_weights, and we will avoid replicated code as much as possible.

Reply:
Broke up the method into `compute_offsets` and `compute_weights`. Added a TODO to move part of the code that is common to multislit and echelle in the parent class